tests/CppUTest/UtestTest.cpp

```diff
@@ -120,6 +120,42 @@ TEST(UtestShell, ExitLeavesQuietly)
     LONGS_EQUAL(0, fixture.getFailureCount());
 }
 
+static bool cpputestHasCrashed;
+
+static void crashMethod()
+{
+    cpputestHasCrashed = true;
+}
+
+TEST(UtestShell, FailWillNotCrashIfNotEnabled)
+{
+    cpputestHasCrashed = false;
+    UtestShell::setCrashMethod(crashMethod);
+
+    fixture.setTestFunction(_failMethod);
+    fixture.runAllTests();
+
+    CHECK_FALSE(cpputestHasCrashed);
+
+    UtestShell::resetFailBehaviour();
+    UtestShell::resetCrashMethod();
+}
+
+TEST(UtestShell, FailWillCrashIfEnabled)
+{
+    cpputestHasCrashed = false;
+    UtestShell::setCrashOnFail();
+    UtestShell::setCrashMethod(crashMethod);
+
+    fixture.setTestFunction(_failMethod);
+    fixture.runAllTests();
+
+    CHECK(cpputestHasCrashed);
+
+    UtestShell::resetFailBehaviour();
+    UtestShell::resetCrashMethod();
+}
+
```

Comment:
Please don't copy-paste, challenge every line, and write tight code.

The "resetFailBehavior" in the first test seems not needed. The nonCrashing test will need to check the test fails (as you kept the NormalTermintor there)

Last request before I merge this. Can you rename "resetFailBehavior" to "restoreDefaulTestTerminator" or something like that. Reason: For symmetry, a reset ought to also have a set (like the crashMethod), but the method isn't called "resetCrashOnFail" and therefore it is probably to not call it reset at all.